[BUG] EGX integration test_collect_list_reductions failures

**Describe the bug**
Tests are failing on 22.06 IT yarn and standalone:

```
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Short]
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Long]
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Boolean]
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Timestamp]
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Decimal(7,3)]
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Decimal(12,2)]
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Array(Byte)]
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Array(Short)]
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Array(Integer)]
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Array(Null)]
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Array(Decimal(7,3))]
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Array(Decimal(12,2))]
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Array(Array(Short))]
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Array(Array(String))]
07:47:23  FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_collect_list_reductions[Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])]
```

`07:47:23  E           AssertionError: GPU and CPU int values are different at [0, 'collect_list(a)', 0, 'child0']
`

--- a/integration_tests/src/main/python/hash_aggregate_test.py
+++ b/integration_tests/src/main/python/hash_aggregate_test.py
@@ -1088,7 +1088,8 @@ def test_arithmetic_reductions(data_gen):
                          ids=idfn)
 def test_collect_list_reductions(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
-        lambda spark: unary_op_df(spark, data_gen).selectExpr('collect_list(a)'),
+        # coalescing because collect_list is not deterministic
+        lambda spark: unary_op_df(spark, data_gen).coalesce(1).selectExpr('collect_list(a)'),
         conf=_no_nans_float_conf)
 
 _struct_only_nested_gens = [all_basic_struct_gen,
@@ -1631,7 +1632,7 @@ def test_count_fallback_when_ansi_enabled(data_gen):
 def test_no_fallback_when_ansi_enabled(data_gen):
     def do_it(spark):
         df = gen_df(spark, [('a', data_gen), ('b', data_gen)], length=100)
-        # coalescing because of first/last are not deterministic
+        # coalescing because first/last are not deterministic
         df = df.coalesce(1).orderBy("a", "b")
         return df.groupBy('a').agg(f.first("b"), f.last("b"), f.min("b"), f.max("b"))
 

